store.rb

```diff
@@ -118,4 +153,26 @@ def get_node(node)
       raise e
     end
   end
+
+  # immediately update cache if node joins/leaves
+  def watch()
+    return if @zk.nil?
+    @root_watcher ||= @zk.register("/", :only => :child) do |event|
```

Comment:
Oh, Ben was right. We'd both change `||=` to `=` and make sure `watch()` is called in the register callback.